universal-application-tool-0.0.1/app/repository/AmazonS3Client.java

```diff
@@ -0,0 +1,108 @@
+package repository;
+
+import com.typesafe.config.Config;
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.inject.ApplicationLifecycle;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+@Singleton
+public class AmazonS3Client {
+  public static final String AWS_S3_REGION = "aws.s3.region";
+  public static final String AWS_S3_BUCKET = "aws.s3.bucket";
+  private static final Logger log = LoggerFactory.getLogger("s3client");
+
+  private final ApplicationLifecycle appLifecycle;
+  private final Config config;
+  private Region region;
+  private String bucket;
+  private S3Client s3;
+
+  @Inject
+  public AmazonS3Client(ApplicationLifecycle appLifecycle, Config config) {
+    this.appLifecycle = appLifecycle;
+    this.config = config;
+
+    log.info("aws s3 enabled: " + String.valueOf(enabled()));
+    if (enabled()) {
+      connect();
+      putTestObject();
+      getTestObject();
+    }
+
+    appLifecycle.addStopHook(
+        () -> {
+          if (s3 != null) {
+            s3.close();
+          }
+          return CompletableFuture.completedFuture(null);
+        });
+  }
+
+  public boolean enabled() {
+    return (config.hasPath(AWS_S3_REGION) && config.hasPath(AWS_S3_BUCKET));
```

Comment:
Question: is there a non-error case where this would be called and return false?

Reply:
These config are set by passing in environment variables and we do not provide default values. So it is possible to run locally without connecting to S3. This allows us to run dev server without providing AWS access keys/token every time. Do we want to have a test bucket that we connect to in dev server? Should we hardcode that bucket name in our default values?